### PR TITLE
nested text onPress not working on last character

### PR DESCRIPTION
**Pull request by fabriziobertoglio1987**
## Summary

This issue fixes https://github.com/facebook/react-native/issues/22747 nested text does not allow you to press on the last character. 

The method reactTagForTouch filters touches based on coordinates x and y. Nested Texts are converted into Spans on Android
https://github.com/facebook/react-native/blob/28fb41a0ab48cc01d606b64744c84e2ac3805f3f/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactBaseTextShadowNode.java#L111-L112
https://developer.android.com/guide/topics/text/spans

reactTagForTouch iterates over the span and triggers the onPress handler if the x,y coordinates correspond to one of the span characters.

## Changelog

[Android] [Fixed] - Nested Text Android onPress does not work with last character

## Test Plan

This changes fix the Java API which can not be tested as explained in commit https://github.com/facebook/react-native/commit/709a441ecf54cd9465f5946af0454ee7d10d5cbe
The java TextTest was excluded from the test suite in commit https://github.com/facebook/react-native/commit/709a441ecf54cd9465f5946af0454ee7d10d5cbe as they need the Yoga libraries to run

**<details><summary>TEST - Clicking on the last letter triggers the callback</summary>**
<p>

Clicking on the last letter does not invoke the onPress callback (in this case a console.warn)

| **BEFORE** | 
|:-------------------------:|
|  |

Clicking on the last letter does invoke the onPress callback (in this case a console.warn)

| **AFTER** | 
|:-------------------------:|
|  |

</details>
</p>

**<details><summary>TEST - Adding and removing Text</summary>**
<p>

<video src="https://user-images.githubusercontent.com/24992535/107541305-48dc6780-6bc7-11eb-8d57-a8aeb57a6879.mp4" />

</details>
</p>

**<details><summary>TEST - Different type of languages</summary>**
<p>

<video src="https://user-images.githubusercontent.com/24992535/107541683-affa1c00-6bc7-11eb-8630-22c2ba4d0973.mp4" />

</details>
</p>

**<details><summary>TEST - Testing other Examples that use onPress handler</summary>**
<p>

<video src="https://user-images.githubusercontent.com/24992535/107541972-f9e30200-6bc7-11eb-9759-6ff9d52bba15.mp4" />

</details>
</p>

**<details><summary>TEST - Text with Inline Images</summary>**
<p>

| Inline View | Inline Image is clipped | 
|:-------------------------:|:-------------------------:|
|  |  |

</details>
</p>

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextView.java [hunk 1]
```diff
@@ -359,7 +359,7 @@ public int reactTagForTouch(float touchX, float touchY) {
         for (int i = 0; i < spans.length; i++) {
           int spanStart = spannedText.getSpanStart(spans[i]);
           int spanEnd = spannedText.getSpanEnd(spans[i]);
-          if (spanEnd > index && (spanEnd - spanStart) <= targetSpanTextLength) {
+          if (spanEnd >= index && (spanEnd - spanStart) <= targetSpanTextLength) {
```
Comment (fabriziobertoglio1987) [inline ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextView.java:362]: the main question is why it did work on non-nested Text. 
I need to find the reason spanEnd reaches the index value on the last characted when the Text is nested in another Text

**Comment by fabriziobertoglio1987** [comment]
I investigated the current failures in the CI and they are related to this pr. I tried find a solution to the failures, which relate to AppCompat (`Error inflating class com.android.internal.widget.ActionBarContextView`)

Unluckily the Java Unit Test for ReactTextTest were disable for an issue with Yoga linking t14964130 

https://github.com/facebook/react-native/blob/b113f665fb6b383cfb8647f180d819918bb52c4b/ReactAndroid/src/test/java/com/facebook/react/views/BUCK#L5-L6

Related Discussion
https://github.com/facebook/react-native/pull/11570
https://github.com/facebook/react-native/pull/11511#issuecomment-268047095

I will invest my time writing Java tests for other functionalities as this seems pretty hard issues to solve. Thanks

File: ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextView.java (hunk 1 above)
Comment (fabriziobertoglio1987) [reply ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextView.java:362]: The issue was not experienced on non-nested text as normal Text will not generate Spans. Spans are generated by nested Text react tags

**Comment by parveen-bhatia** [comment]
Facing same issue. What are the plans to move it forward ?

**Pull request closed without merging**